subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -38,40 +266,83 @@ static int download_client_callback(const struct download_client_evt *event)
 			callback(FOTA_DOWNLOAD_EVT_ERROR);
 			return err;
 		}
+
 		if (size > PM_MCUBOOT_SECONDARY_SIZE) {
 			LOG_ERR("Requested file too big to fit in flash\n");
 			callback(FOTA_DOWNLOAD_EVT_ERROR);
 			return -EFBIG;
 		}
 
-		err = flash_img_buffered_write(&flash_img,
-				(u8_t *)event->fragment.buf,
-				event->fragment.len, false);
-		if (err != 0) {
-			LOG_ERR("flash_img_buffered_write error %d", err);
+		if (first_fragment) {
+			first_fragment = false;
+			image_type = check_img_type((u8_t *) event->fragment.buf
+						    , event->fragment.len);
+		}
+
+		if (image_type == MCUBOOT) {
+			err = mcuboot_write(&flash_img,
+					    (u8_t *)event->fragment.buf,
+					    event->fragment.len);
+			if (err != 0) {
+				LOG_ERR("flash_img_buffered_write error %d",
```

Comment:
Shouldn't we disconnect in case of MCUBOOT write error as well? (in the same way as in case of modem FW).

Reply:
Would probably be a good idea I'll add it.